plugins/headlamp-plugin/bin/headlamp-plugin.js

```diff
@@ -231,6 +233,71 @@ function extract(pluginPackagesPath, outputPlugins) {
   return 0;
 }
 
+/**
+ * Creates a tarball of the plugin package. The tarball is placed in the outputFolderPath.
+ * It moves files from:
+ *   packageName/dist/main.js to packageName/main.js
+ *   packageName/package.json to packageName/package.json
+ * And then creates a tarball of the resulting folder.
+ *
+ * @param {string} pluginDir - path to the plugin package.
+ * @param {string} outputFolderPath - folder where the tarball is placed.
+ *
+ * @returns {0 | 1} Exit code, where 0 is success, 1 is failure.
+ */
+async function createArchive(pluginDir, outputFolderPath) {
+  const pluginPath = path.resolve(pluginDir);
+  if (!fs.existsSync(pluginPath)) {
+    console.error(`"${pluginPath}" does not exist. Not creating archive.`);
+    return 1;
+  }
+
+  if (!fs.existsSync(outputFolderPath)) {
```

Comment:
Maybe some of this code could be shared with code in extract(). This could be done later.